_upgrading/sopel-8.md

```diff
@@ -0,0 +1,420 @@
+comparing and normalizing nicknames itself.
+
+To support handling this under the hood, the bot's `config.core.nick` setting is
+now stored and returned as a normal `str`. It and any other strings that you'd
+like to treat as nicknames should be passed through the `bot.make_identifier()`
+method to get the same `Identifier` type you're used to from Sopel 7, configured
+to use the IRC server's advertised casemapping method.
+
+### Time-handling changes
+
+`trigger.time` is now an "aware" `datetime` object, meaning it has a UTC offset
+associated with it. Comparison or arithmetic operations between "aware" and
+"naive" `datetime` objects are not allowed; code that manipulates `trigger.time`
+values will need to be updated for Sopel 8.
+
+The fallback format string used by `sopel.tools.time.format_time()` if no other
+source provides one changed from outputting the timezone name (`UTC`) to the UTC
+offset (`+0000`).
+
+`format_time()`'s handling of "aware" and "naive" `datetime`s was also improved,
+but those changes should be transparent to both users and plugin developers.
+
+`sopel.tools.time.validate_timezone()` now also raises a `ValueError` even if
+the timezone to be validated is `None`. This was previously a special case that
+_returned_ `None`.
+
+### Database changes
+
+`bot.db.get_nick_id()` no longer creates a new ID by default if its `create`
+parameter is unspecified.
+
+### IRC event changes
+
+The event names `RPL_INVITELIST` and `RPL_ENDOFINVITELIST` have been updated per
+clarifications from the living-specification project at modern.ircdocs.horse.
+`RPL_INVEXLIST` and `RPL_ENDOFINVEXLIST` were added to the list of named events
+that Sopel knows about.
+
+Plugin code might need to be updated as a result of these changes, but keep in
+mind that [different IRC servers mean different things][invitelist-madness] when
+they send these events. If in doubt, specify a raw numeric value (e.g. `'346'`).
+Writing truly cross-network plugins that react to these events should be
+undertaken _very_ carefully.
+
+  [invitelist-madness]: https://github.com/ircdocs/modern-irc/issues/42
+
+### IRC connection status monitoring
+
+Previously a simple `bool` value, `bot.connection_registered` is now a property
+combining status information from across the bot's subsystems to answer the very
+important question, "Is the `bot` actually registered to an IRC network?" That
+is, Sopel not only has a _connection_ open, but the IRC server has _accepted_
+the bot as a _client_ and it's possible to _send commands_ to the IRC server.
+
+This is useful to plugins with code that runs without a triggering IRC event but
+still outputs to IRC, such as a `@sopel.plugin.interval()` job or acting on data
+received [via a socket][sopel-sockmsg]. `if not bot.connection_registered`, the
+output can be skipped, retried after a delay, etc.
+
+_Hopefully no one was overwriting the old simple attribute in plugin code, but
+now they **can't** do that._
+
```

Comment:
To go along with [my other suggestion](https://github.com/sopel-irc/sopel.chat/pull/46/files#r1564816742), downplaying the fact that it used to be writable by plugins:

```suggestion
```